go-controller/pkg/ovn/master.go

```diff
@@ -747,27 +747,122 @@ func (oc *Controller) syncNodeClusterRouterPort(node *kapi.Node, hostSubnets []*
 		}
 	}
 
+	// TODO(flaviof): delete this!!
+
+	// lrpName := types.RouterToSwitchPrefix + node.Name
+	// lrpArgs := []string{
+	// 	"--if-exists", "lrp-del", lrpName,
+	// 	"--", "lrp-add", types.OVNClusterRouter, lrpName, nodeLRPMAC.String(),
+	// }
+	// for _, hostSubnet := range hostSubnets {
+	// 	gwIfAddr := util.GetNodeGatewayIfAddr(hostSubnet)
+	// 	lrpArgs = append(lrpArgs, gwIfAddr.String())
+	// }
+	// if config.Gateway.Mode != config.GatewayModeLocal {
+	// 	// "local" mode requires NAT on the cluster router, which is not yet supported yet when
+	// 	// multiple DGPs are on the same router, so we can't set the gateway-chassis here.
+	// 	lrpArgs = append(lrpArgs, "--", "lrp-set-gateway-chassis", lrpName, chassisID, "1")
+	// }
+	// _, stderr, err := util.RunOVNNbctl(lrpArgs...)
+	// if err != nil {
+	// 	klog.Errorf("Failed to add logical port to router, stderr: %q, error: %v", stderr, err)
+	// 	return err
+	// }
+
 	lrpName := types.RouterToSwitchPrefix + node.Name
-	lrpArgs := []string{
-		"--if-exists", "lrp-del", lrpName,
-		"--", "lrp-add", types.OVNClusterRouter, lrpName, nodeLRPMAC.String(),
-	}
+	lrpNetworks := []string{}
 	for _, hostSubnet := range hostSubnets {
 		gwIfAddr := util.GetNodeGatewayIfAddr(hostSubnet)
-		lrpArgs = append(lrpArgs, gwIfAddr.String())
+		lrpNetworks = append(lrpNetworks, gwIfAddr.String())
+	}
+	logicalRouterPort := nbdb.LogicalRouterPort{
+		Name:     lrpName,
+		MAC:      nodeLRPMAC.String(),
+		Networks: lrpNetworks,
+	}
+	logicalRouter := nbdb.LogicalRouter{}
+	opModels := []libovsdbops.OperationModel{
+		{
+			Model: &logicalRouterPort,
+			ModelPredicate: func(lrp *nbdb.LogicalRouterPort) bool {
+				return lrp.Name == lrpName
+			},
+			OnModelUpdates: []interface{}{
+				&logicalRouterPort.Name,
+				&logicalRouterPort.MAC,
+				&logicalRouterPort.Networks,
+			},
+			DoAfter: func() {
+				if logicalRouterPort.UUID != "" {
+					logicalRouter.Ports = []string{logicalRouterPort.UUID}
+				}
+			},
+		},
+		{
+			Model:          &logicalRouter,
+			ModelPredicate: func(lr *nbdb.LogicalRouter) bool { return lr.Name == types.OVNClusterRouter },
+			OnModelMutations: []interface{}{
+				&logicalRouter.Ports,
+			},
+			ErrNotFound: true,
+		},
+	}
+
+	// FIXME(flaviof): should use ops to potentially combine gateway chassis creation below
```

Comment:
After talking with @jcaamano and then @alexanderConstantinescu , we are leaving this as is for this go around.

The reason being, (1)we are aiming to just keep it compatible with the previous implementation at this point.
Also, (2)it would require a new functionality in libovsdbops.OperationModel to handle mutation on an object (lrp) that
does not yet exist prior to the transaction.